Jenkinsfile

```diff
@@ -0,0 +1,99 @@
+        string(name:"PORT_NO", defaultValue:"8989", description: "Bind Port Number")
+
+        booleanParam(name: 'BUILD', defaultValue: false, description: 'Check To Build')
+        booleanParam(name: 'SONAR_ANALYSIS', defaultValue: false, description: 'Check To Sonar Analysis')
+        // booleanParam(name: 'TEST', defaultValue: false, description: 'Check To Test')
+        booleanParam(name: 'PUBLISH', defaultValue: false, description: 'Check To Publish')
+        booleanParam(name: 'DOCKER_BUILD', defaultValue: false, description: 'Check To DOCKER_BUILD')
+        booleanParam(name: 'DOCKER_HUB_PUBLISH', defaultValue: false, description: 'Check To DOCKER_HUB_PUBLISH')
+        booleanParam(name: 'DOCKER_RUN', defaultValue: false, description: 'Check To DOCKER_RUN')
+    }
+
+    stages {
+        stage('build') {
+            when {
+                expression { params.BUILD || params.PUBLISH || params.TEST}
+            }
+            steps {
+                bat '''dotnet build %SOLUTION_NAME%'''
+            }
+        }
+
+        
+        stage('sonar') {
+             when {
+                expression { return params.SONAR_ANALYSIS}
+            }
+            steps{
+                bat """
+                        dotnet ${SONAR_MS_TOOL}  begin /k:"%SONAR_PROJECT_NAME%" /d:sonar.host.url=${SONAR_URL}  /d:sonar.login="${SONAR_TOKEN}"
+                        dotnet  build
+                        dotnet ${SONAR_MS_TOOL} end  /d:sonar.login="${SONAR_TOKEN}"
+                    """
+            }     
+        }
+
+        stage('publish') {
+             when {
+                expression { return params.PUBLISH || params.DEPLOY }
+            }
+            steps {
+                bat '''dotnet publish %SOLUTION_NAME% -p:Configuration=release -v:q -o ../artifacts'''
+            }
+        }
+
+        
+
+
+         stage('Docker build') {
+            when {
+                  expression {return params.DOCKER_BUILD}
+            }
+             steps{
+                bat 'docker build -t %DOCKER_IMAGE_NAME% .'
+             }
+        }
+
+        stage('UpdateDockerRepo') {
+            when {
+                  expression {return params.DOCKER_HUB_PUBLISH}
+            }
+            
+            steps {
+             script{
+                    docker.withRegistry('','docker_creds')
+                    {
+                        
+                         bat ''' docker tag %DOCKER_IMAGE_NAME%:latest subtleparesh/%DOCKER_IMAGE_NAME% 
+                        docker push subtleparesh/%DOCKER_IMAGE_NAME%:latest'''
+                    }
+                }
+            }
+        }
+         stage('run docker image'){
+             when {
+                  expression {return params.DOCKER_RUN}
+            }
+            steps{
+                echo 'run the image'
+                bat 'docker run -p %PORT_NO%:80 -e SOLUTION_DLL=%SOLUTION_DLL%  %DOCKER_IMAGE_NAME%'
```

Comment:
container port is should be parameterized and docker image should be pulled from docker repo.